### PR TITLE
Improve fixup session handling and disposal

### PR DESCRIPTION
**Pull request by mkondratek**
Possible a step towards fixing https://github.com/sourcegraph/cody-issues/issues/314.

## Test plan
1. Try to get to the "invalid state" when the inline actions are disabled

### CODE REVIEW

File: src/main/kotlin/com/sourcegraph/cody/agent/CodyAgentService.kt [hunk 1]
```diff
@@ -250,6 +254,7 @@ class CodyAgentService(private val project: Project) : Disposable {
               try {
                 callback.accept(agent)
               } catch (e: Exception) {
+                logger.warn(e)
```
Comment (mkondratek) [inline src/main/kotlin/com/sourcegraph/cody/agent/CodyAgentService.kt:257]: The error: `IllegalStateException("Cannot start new session when one is already active")` was there but we ignored it 🙃

File: src/main/kotlin/com/sourcegraph/cody/edit/sessions/FixupSession.kt [hunk 2]
```diff
@@ -411,6 +411,7 @@ abstract class FixupSession(
     if (project.isDisposed) return
     performedActions.forEach { it.dispose() }
     editor.document.removeDocumentListener(documentListener)
+    cancellationToken.dispose()
```
Comment (mkondratek) [inline src/main/kotlin/com/sourcegraph/cody/edit/sessions/FixupSession.kt:403]: so the controller unsets active session

Comment (pkukielka) [reply src/main/kotlin/com/sourcegraph/cody/edit/sessions/FixupSession.kt:403]: We need a bit of additional cleanup:
1. On `finish` and `error` state we now call `cancellationToken.dispose()`
We should either call `dispose()`, or we should move all this actions to `cancellationToken.onFinished` in the constructor.
2. There is also `fun dismiss()` which looks like incomplete `dispose` now. I guess we should delete it and replace with `dispose()`?

Comment (mkondratek) [reply src/main/kotlin/com/sourcegraph/cody/edit/sessions/FixupSession.kt:403]: fixed

File: src/main/kotlin/com/sourcegraph/cody/edit/sessions/FixupSession.kt [hunk 3]
```diff
@@ -88,14 +88,6 @@ abstract class FixupSession(
         } catch (x: Exception) {
           logger.debug("Session cleanup error", x)
         }
-
-        runInEdt {
-          try { // Disposing inlay requires EDT.
-            Disposer.dispose(this)
```
Comment (mkondratek) [inline src/main/kotlin/com/sourcegraph/cody/edit/sessions/FixupSession.kt:94]: the sessions calls `cancelationToken.dispose()` now

File: src/main/kotlin/com/sourcegraph/cody/edit/sessions/FixupSession.kt [hunk 4]
```diff
@@ -411,6 +399,8 @@ abstract class FixupSession(
     if (project.isDisposed) return
     performedActions.forEach { it.dispose() }
     editor.document.removeDocumentListener(documentListener)
```
Comment (mkondratek) [inline src/main/kotlin/com/sourcegraph/cody/edit/sessions/FixupSession.kt:401]: @pkukielka, the problem was that we called it twice - we should not see more errors from this one

File: src/main/kotlin/com/sourcegraph/cody/edit/sessions/FixupSession.kt (hunk 3 above)
Comment (pkukielka) [reply src/main/kotlin/com/sourcegraph/cody/edit/sessions/FixupSession.kt:94]: Cool!

File: src/main/kotlin/com/sourcegraph/cody/edit/sessions/FixupSession.kt (hunk 4 above)
Comment (pkukielka) [reply src/main/kotlin/com/sourcegraph/cody/edit/sessions/FixupSession.kt:401]: Great :)

**Review by pkukielka — APPROVED** [review]
LGTM

**Pull request merged**